mkctr.go

```diff
@@ -371,7 +371,21 @@ func createImageForBase(bp *buildParams, logf logf, base v1.Image, platform v1.P
 		logf("output %v -> %v", gp, n)
 		files[n] = dst
 	}
-	layer, err := layerFromFiles(logf, files)
+	// Determine media type of the base image.
+	var layerMediaType types.MediaType
+	mt, err := base.MediaType()
+	if err != nil {
+		return nil, fmt.Errorf("error determining base image media type: %v", err)
+	}
+	switch mt {
+	case types.OCIManifestSchema1:
+		layerMediaType = types.OCILayer
+	case types.DockerManifestSchema2:
+		layerMediaType = types.DockerLayer
+	default:
+		return nil, fmt.Errorf("unknown base image media type %v, accepted types are OCI image index and docker manifest list", mt)
```

Comment:
```suggestion
		return nil, fmt.Errorf("unknown base image media type %v, accepted types are OCI image index (%s) and docker manifest list (%s)", mt, types.OCIManifestSchema1, types.DockerManifestSchema2)
```

Reply:
This was actually incorrectly referring to OCI image index and Docker manifest list, I've fixed it now.